### PR TITLE
feat(lane_change): check prepare phase in turn direction lanes

### PR DESCRIPTION
**Pull request by zulfaqar-azmi-t4**
## Description

⚠️ Need to merge together with the following PR
https://github.com/autowarefoundation/autoware_launch/pull/943

Make lane change check more stricter in when performing lane change after making a turn.
This is done by performing collision check in the prepare phase.

## Before PR

https://github.com/autowarefoundation/autoware.universe/assets/93502286/6155e1f3-7bc7-4da1-886d-919fbcf6c4c2

## After PR

https://github.com/autowarefoundation/autoware.universe/assets/93502286/6437f693-f734-4c17-8c7e-4e1eae64a61d

## Tests performed

Using scenario provided in [TIER IV internal link](https://tier4.atlassian.net/browse/RT0-30817)

Evaluator [TIER IV internal link](https://evaluation.tier4.jp/evaluation/reports/1a2527bf-3c4b-5b24-9f7f-26034bbc6b9e?project_id=prd_jt). There seems to be no issue regarding this.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by shmpwk** [comment]
@Owen-Liuyuxuan 
Could you review this?

**Review by Owen-Liuyuxuan — APPROVED** [review]
Thanks for the response and updates

File: planning/behavior_path_lane_change_module/src/utils/utils.cpp [hunk 1]
```diff
@@ -30,13 +30,16 @@
 #include <motion_utils/trajectory/interpolation.hpp>
 #include <motion_utils/trajectory/path_with_lane_id.hpp>
 #include <motion_utils/trajectory/trajectory.hpp>
+#include <rclcpp/logger.hpp>
```
[the remark this answers is not in the document]
Comment (Owen-Liuyuxuan) [reply planning/behavior_path_lane_change_module/src/utils/utils.cpp]: solved

File: planning/behavior_path_lane_change_module/README.md [hunk 2]
```diff
@@ -555,6 +555,7 @@ The following parameters are configurable in [lane_change.param.yaml](https://gi
...
 | `enable_collision_check_for_prepare_phase.general_lanes` | [-]   | boolean | Perform collision check starting from the prepare phase for situations not explicitly covered by other settings (e.g., intersections). If `false`, collision check only evaluated for lane changing phase. | false         |
 | `enable_collision_check_for_prepare_phase.intersection`  | [-]   | boolean | Perform collision check starting from prepare phase when ego is in intersection. If `false`, collision check only evaluated for lane changing phase.                                                       | false         |
+| `enable_collision_check_for_prepare_phase.turns`         | [-]   | boolean | Perform collision check starting from prepare phase when ego is in lanelet with turn direction tags. If `false`, collision check only evaluated for lane changing phase.                                   | false         |
```
[the remark this answers is not in the document]
Comment (Owen-Liuyuxuan) [reply planning/behavior_path_lane_change_module/README.md]: solved

**Review by shmpwk — APPROVED** [review]
Thank you!

**Pull request merged**